### ISSUE TITLE
HydraNorth/Fedora4 Basic Authorization

### ISSUE THREAD

**Issue by pgwillia** [ualbertalib/HydraNorth#132]
Basic Authorization would mean that the application would have to be logged in as an admin before being able to access API-M for any of the add/modify tasks

In general, you enable basic authentication for Fedora4 by following [these instructions](https://wiki.duraspace.org/display/FEDORA40/How+to+Configure+Servlet+Container+Authentication). Then in HydraNorth the fedoraAdmin username and password assigned would go into [config/fedora.yml](https://github.com/ualbertalib/HydraNorth/blob/master/config/fedora.yml).

**Comment by pbinkley** [comment]
When we're testing this, let's check whether the Hydra user is credited
with changes, as in current Hydra North, or whether the Fedora user gets
all the credit, as in old ERA. We definitely want the former.

Peter

Peter Binkley
Digital Initiatives Technology Librarian
Information Technology Services
peter.binkley@ualberta.ca

2-10K Cameron Library
University of Alberta
Edmonton, Alberta
Canada T6G 2J8

phone 780-492-3743
fax 780-492-9243

**Comment by pgwillia** [comment]
It does not record the Fedora user. bypassAdmin is credited with changes. The application (HydraNorth) independently records changes and depositor in it's own namespace.

**Comment by pgwillia** [comment]
The Hydra user is not credited with changes in current HydraNorth.

**Comment by pbinkley** [comment]
Hmm - I looked at that last week. They're credited with ownership at the
datastream level, at least, isn't it? I'm checking again.

Peter

Peter Binkley
Digital Initiatives Technology Librarian
Information Technology Services
peter.binkley@ualberta.ca

2-10K Cameron Library
University of Alberta
Edmonton, Alberta
Canada T6G 2J8

phone 780-492-3743
fax 780-492-9243

**Comment by pbinkley** [comment]
Thing I learned: if you shut down a Vagrant vm from inside with shutdown,
it goes down but Vagrant thinks it's still up, which leads to confusion. At
least that seems to be state I got into; vagrant halt had to force the
machine down. Won't do that again.

Peter

Peter Binkley
Digital Initiatives Technology Librarian
Information Technology Services
peter.binkley@ualberta.ca

2-10K Cameron Library
University of Alberta
Edmonton, Alberta
Canada T6G 2J8

phone 780-492-3743
fax 780-492-9243
